eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php

```diff
@@ -1561,4 +1544,66 @@ private function getAllLocationsQueryBuilder(array $columns)
 
         return $query;
     }
+
+    /**
+     * Create QueryBuilder for selecting node data.
+     *
+     * @param array|null $translations Filters on language mask of content if provided.
+     * @param bool $useAlwaysAvailable Respect always available flag on content when filtering on $translations.
+     *
+     * @return \Doctrine\DBAL\Query\QueryBuilder
+     */
+    private function createNodeQueryBuilder(array $translations = null, bool $useAlwaysAvailable = true): QueryBuilder
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $queryBuilder
+            ->select('t.*')
+            ->from('ezcontentobject_tree', 't');
+
+        if (!empty($translations)) {
+            $dbPlatform = $this->connection->getDatabasePlatform();
+            $expr = $queryBuilder->expr();
+            $mask = $this->generateLanguageMaskFromLanguageCodes($translations, $useAlwaysAvailable);
+
+            $queryBuilder->innerJoin(
+                't',
+                'ezcontentobject',
+                'c',
+                $expr->andX(
+                    $expr->eq('t.contentobject_id', 'c.id'),
+                    // Won't work on Oracle, consider contribute bitwise features to query builder, and
+                    // contribute support for it handling DBM differences. Or detect Oracle here and do bitand()
```

Comment:
drop this now that code actually uses `dbPlatform->getBitAndComparisonExpression `